network/src/peers/peer/outbound_handler.rs

```diff
@@ -38,49 +44,44 @@ pub(super) enum PeerAction {
 #[derive(Clone, Debug)]
 pub struct PeerHandle {
     pub(super) sender: mpsc::Sender<PeerAction>,
+    pub queued_outbound_message_count: Arc<AtomicUsize>,
```

Comment:
i dont like delta metrics because they can get out of sync. i don't see a way to solve this with tokio channels though... is this metric useful? it seemed useful when we had one channel to rule them all, but it increasingly looks to me like maybe we want a payload emission at app layer -> payload emitted on network latency metric? would also be way easier to implement and not error prone.

Reply:
i think this metric would be more useful for identifying actual latency problems because it tracks each payload individually -- however if there is disagreement, we should just make a wrapped `Sender/Receiver` type that can more tightly couple the metric into the channel (and be used in several places)

Reply:
when running in bootnode mode and with a few hundred connections, we are actually constantly at a combined depth of all the outbound queues greater than 1k items, which might indicate that we're slowly reaching practical connection limits (at least when syncing at the same time); we need to know if we're able to push out all the messages quickly, otherwise we might be growing memory or become swamped and reply slowly.

> it seemed useful when we had one channel to rule them all

we've always had individual per-peer outbound channels (or at least ever since we've had metrics), so the same principle applies

Reply:
that being said, and even though I saw no divergence with a lot of connection changes (meaning this is an improvement to what we currently have), I agree that the `queues::OUTBOUND` setup proposed in this PR is not exactly trivial, and I'm open to alternatives that will allow us to monitor how many items we have waiting to be sent to peers

Reply:
does `metrics` crate/prometheus have tagged metrics? would probably make the data more useful as well, so we can see concrete metrics on a per-connection basis. latency metrics also sound like they capture this rather well. i'm open to depth as well, and we can just pass this through for now, but the usage of out-of-band metrics like this is not advised.

Reply:
> does metrics crate/prometheus have tagged metrics?

I think it's done via labels: https://prometheus.io/docs/practices/naming/#labels.

> we can see concrete metrics on a per-connection basis

an individual breakdown could be interesting indeed, as some outliers might be causing the combined values to become "weird"

> i'm open to depth as well, and we can just pass this through for now

I'll file a new GitHub issue related to the aforementioned "wholesale" value updates to avoid divergence and one for prometheus labels so we can handle those as follow-ups; as for now, since these PRs do appear to fix (or restore) the values, I propose to merge them as-is to keep the changes small and self-contained